### PR TITLE
DH-1467 Refactor/tidy up oauth controller

### PR DESCRIPTION
**Pull request by feedmypixel**
Update tests for Oauth controller

Originally it sat with #1229, but I have split it out as I'm not sure yet if the work in #1229 is actually needed.

### Of note
all the logging will be going once this fix is confirmed

### CODE REVIEW

File: src/apps/oauth/controllers.js [hunk 1]
```diff
@@ -1,28 +1,11 @@
...
-      client_id: config.oauth.clientId,
-      client_secret: config.oauth.clientSecret,
-      redirect_uri: config.oauth.redirectUri,
-    },
-    json: true,
-  }
-
-  return request(options)
-}
+const { getAccessToken } = require('./services')
```
Comment (teneightfive) [inline src/apps/oauth/controllers.js]: I thought we were trying to remove the idea of `services`? They don't feel well enough defined to me.

File: src/apps/oauth/services.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const request = require('request-promise')
+
+const config = require('./../../../config')
+
+function getAccessToken (oauthCode) {
+  const options = {
+    method: 'POST',
+    url: config.oauth.tokenFetchUrl,
```
Comment (teneightfive) [inline src/apps/oauth/services.js]: `get()` on these properties?

File: src/apps/oauth/services.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const request = require('request-promise')
+
+const config = require('./../../../config')
+
+function getAccessToken (oauthCode) {
```
Comment (teneightfive) [inline src/apps/oauth/services.js]: What about just `code`? Then you can use shorthand in the object definition.

File: src/apps/oauth/controllers.js (hunk 1 above)
Comment (feedmypixel) [reply src/apps/oauth/controllers.js]: I'd disagree, `services` is a pretty well-defined pattern, outside and Inside of our `app`.
screen shot 2018-02-02 at 10 29 51

File: src/apps/oauth/services.js (hunk 2 above)
Comment (feedmypixel) [reply src/apps/oauth/services.js]: These are all required for Oauth to work so I don't think `get()` is needed here

File: test/unit/apps/oauth/controllers.test.js [hunk 4]
```diff
@@ -18,12 +26,20 @@ describe('OAuth controller', () => {
       session: {},
     })
     this.nextSpy = sandbox.spy()
+
+    set(this.mockConfig, 'oauth', {
```
Comment (teneightfive) [inline test/unit/apps/oauth/controllers.test.js]: Wonder whether set is overkill here? You've defined the `mockConfig` property above so can be sure it exists. You're also not setting a deep property so don't think you'll need this helper.

File: test/unit/apps/oauth/controllers.test.js [hunk 5]
```diff
@@ -18,12 +26,20 @@ describe('OAuth controller', () => {
...
+      tokenFetchUrl: this.mockFetchUrl.host + this.mockFetchUrl.path,
+      clientSecret: 'mock-client-secret',
+      redirectUri: '/mock/callback/url',
+      clientId: 'mockClientId',
+    })
   })
 
   describe('#redirectOAuth', () => {
+    this.mockUUIDvalue = 'mock-uuid-1234'
```
Comment (teneightfive) [inline test/unit/apps/oauth/controllers.test.js]: Not sure if this has access to the `this` that you want to set this property on. Might be worth double checking this works as expected.

File: test/unit/apps/oauth/controllers.test.js [hunk 6]
```diff
@@ -116,30 +122,21 @@ describe('OAuth controller', () => {
...
-            path: '/example-fetch-token-url',
-          }
-
-          set(this.mockConfig, 'oauth', {
-            tokenFetchUrl: this.mockFetchUrl.host + this.mockFetchUrl.path,
-            clientSecret: 'mock-client-secret',
-            redirectUri: '/mock/callback/url',
-            clientId: 'mockClientId',
-          })
           set(this.reqMock, 'query', {
```
Comment (teneightfive) [inline test/unit/apps/oauth/controllers.test.js]: Again, not sure you need `set` here as you know `reqMock` property exists.

File: test/unit/apps/oauth/controllers.test.js [hunk 7]
```diff
@@ -161,91 +158,97 @@ describe('OAuth controller', () => {
...
-      }
-
       beforeEach(() => {
-        set(this.mockConfig, 'oauth', {
-          tokenFetchUrl: mockFetchUrl.host + mockFetchUrl.path,
-          clientSecret: 'mock-client-secret',
-          redirectUri: '/mock/callback/url',
-          clientId: 'mockClientId',
-        })
         set(this.reqMock, 'query', {
```
Comment (teneightfive) [inline test/unit/apps/oauth/controllers.test.js]: Same as above re need for `set`.

File: test/unit/apps/oauth/controllers.test.js (hunk 5 above)
Comment (feedmypixel) [reply test/unit/apps/oauth/controllers.test.js]: this is being tested on [line 69](https://github.com/uktrade/data-hub-frontend/pull/1228/files/f7702736f527422e2e5b72fd728927a38d728795#diff-224a60a8afda1ee300bdcdf49c88b4cdR69)

**Pull request merged**